sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -0,0 +1,195 @@
+package certificates
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stackrox/rox/sensor/common/message"
+)
+
+var (
+	log                            = logging.LoggerForModule()
+	ErrCertificateRequesterStopped = errors.New("certificate requester is stopped")
+)
+
+// Requester defines an interface for requesting TLS certificates from Central
+type Requester interface {
+	Start()
+	Stop()
+	RequestCertificates(ctx context.Context) (*Response, error)
+}
+
+// NewLocalScannerCertificateRequester creates a new local scanner certificate requester that communicates through
+// the specified channels and initializes a new request ID for reach request.
+// To use it call Start, and then make requests with RequestCertificates, concurrent requests are supported.
+// This assumes that the returned certificate requester is the only consumer of `receiveC`.
+func NewLocalScannerCertificateRequester(sendC chan<- *message.ExpiringMessage,
+	receiveC <-chan *central.IssueLocalScannerCertsResponse) Requester {
+	return newRequester[
+		*central.IssueLocalScannerCertsRequest,
+		*central.IssueLocalScannerCertsResponse,
+	](
+		sendC,
+		receiveC,
+		&LocalScannerMessageFactory{},
+		&LocalScannerResponseFactory{},
+	)
+}
+
+// NewSecuredClusterCertificateRequester creates a new certificate requester that communicates through
+// the specified channels and initializes a new request ID for reach request.
+// To use it call Start, and then make requests with RequestCertificates, concurrent requests are supported.
+// This assumes that the returned certificate requester is the only consumer of `receiveC`.
+func NewSecuredClusterCertificateRequester(sendC chan<- *message.ExpiringMessage,
+	receiveC <-chan *central.IssueSecuredClusterCertsResponse) Requester {
+	return newRequester[
+		*central.IssueSecuredClusterCertsResponse,
+		*central.IssueSecuredClusterCertsResponse,
+	](
+		sendC,
+		receiveC,
+		&SecuredClusterMessageFactory{},
+		&SecuredClusterResponseFactory{},
+	)
+}
+
+func newRequester[ReqT any, ResT ProtobufResponse](
```

Comment:
One optional nitpick, I cannot help but think about REST when I see `ResT`. Perhaps consider this change?
```suggestion
func newRequester[ReqT any, RespT ProtobufResponse](
```